Application/EdFi.Ods.Api/Container/Modules/ApplicationModule.cs

```diff
@@ -71,141 +73,177 @@ protected override void Load(ContainerBuilder builder)
 
             builder.RegisterType<ApiKeyContextProvider>()
                 .As<IApiKeyContextProvider>()
-                .As<IHttpContextStorageTransferKeys>();
+                .As<IHttpContextStorageTransferKeys>()
+                .SingleInstance();
 
             builder.RegisterType<SchoolYearContextProvider>()
                 .As<ISchoolYearContextProvider>()
-                .As<IHttpContextStorageTransferKeys>();
+                .As<IHttpContextStorageTransferKeys>()
+                .SingleInstance();
 
             // Primary context storage for ASP.NET web applications
             builder.RegisterType<HttpContextStorage>()
-                .As<IContextStorage>();
+                .As<IContextStorage>()
+                .SingleInstance();
 
             // Secondary context storage for background tasks running in ASP.NET web applications
             // Allows selected context to flow to worker Tasks (see IHttpContextStorageTransferKeys and IHttpContextStorageTransfer)
             builder.RegisterType<CallContextStorage>()
-                .As<IContextStorage>();
+                .As<IContextStorage>()
+                .SingleInstance();
 
             // Features to transfer context from HttpContext to the secondary storage in ASP.NET applications
             builder.RegisterType<HttpContextStorageTransfer>()
-                .As<IHttpContextStorageTransfer>();
+                .As<IHttpContextStorageTransfer>()
```

Comment:
This came up from the field PR, but we've got a double register for IContextStorage happening here, which used to mean the first one would win but now means the last one will win. We're running all our requests using CallContextStorage instead of HttpContextStorage at the moment. Can we fix this with this PR?

Reply:
yes good point, I will remove the httpcontext since its not used.